storage/replica_command.go

```diff
@@ -414,9 +414,8 @@ func (r *Replica) EndTransaction(batch engine.Engine, ms *engine.MVCCStats, args
 				}
 			} else if ct.ModifiedSpanTrigger != nil {
 				if ct.ModifiedSpanTrigger.GetSystemDBSpan() {
-					// Check if we need to gossip the system config, but do it in a separate
-					// goroutine, and after the batch has been committed.
-					batch.Defer(func() { go r.maybeGossipSystemConfig() })
+					// Check if we need to gossip the system config.
```

Comment:
I originally put it in a separate one because I wasn't positive it was safe otherwise.
I dug deeper for this PR, and it looks like it shouldn't cause problems.